napari/layers/utils/layer_utils.py

```diff
@@ -684,6 +684,43 @@ def features_to_pandas_dataframe(features: Any) -> pd.DataFrame:
     return features
 
 
+def _features_from_layer(
+    *,
+    features: Optional[Union[Dict[str, np.ndarray], pd.DataFrame]] = None,
+    properties: Optional[Union[Dict[str, np.ndarray], pd.DataFrame]] = None,
+    property_choices: Optional[Dict[str, np.ndarray]] = None,
+    num_data: Optional[int] = None,
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """Coerces a layer's keyword arguments to feature values and defaults tables."""
+    if properties is not None or property_choices is not None:
+        features = _features_from_properties(
+            properties=properties,
+            property_choices=property_choices,
+            num_data=num_data,
+        )
+    else:
+        features = _validate_features(features, num_data=num_data)
+    defaults = pd.DataFrame(
+        {
+            name: _get_column_default_value(column)
+            for name, column in features.items()
+        },
+        index=range(1),
+    )
+    return features, defaults
+
+
+def _get_column_default_value(column: pd.Series) -> Optional:
+    """Get the default value from a feature column."""
+    if column.size > 0:
+        return column.iloc[-1]
+    if isinstance(column.dtype, pd.CategoricalDtype):
+        choices = column.dtype.categories
+        if choices.size > 0:
+            return choices[0]
```

Comment:
Should this be nested in the `if` above? I'm not super familiar with pandas categorical type, but I would assume a categorical column that has values would still have size > 0.